### PR TITLE
Cst jack statistical outlier 

### PR DESCRIPTION
**Pull request by jackchallen**
Weka stats can provide some useful values. It's sensible to go
    looking at these values for enormous outliers. There are a few
    methods to calculate outliers, but there's no persistently good
    measure that can be applied to all of our statistics.
    I've looked at (and tested) interquartile range and
    the Jarque-Bera method, but both of these perform horribly with
    say PUMPS_TXQ_FULL - producing values of 100-1e7 across a few
    different clusters, and there's no sane way to assess that.
    
    I don't think there's going to be a single correct answer that
    covers all of our statistics, but for now I've just gone with
    looking at a multiplier of the standard deviation. In the case
    of PUMPS_TXQ_FULL, a value that's 10x outside the standard
    deviation of all other backends is quite possibly an outlier indeed.
    
    This will probably need tuning.

### CODE REVIEW

File: scripts.d/ta/780_statistical_outlier.sh [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+for STATISTIC_TO_CHECK in "${CONTAINER_LEVEL_STATISTICS[@]}" ; do
+    STATISTIC_NAME=$( echo ${STATISTIC_TO_CHECK} | awk -F';' '{print $1}')
+    INTERVAL=$(       echo ${STATISTIC_TO_CHECK} | awk -F';' '{print $2}')
+    MAX_STDDEV=$(     echo ${STATISTIC_TO_CHECK} | awk -F';' '{print $3}')
+    WTA_REFERENCE=$(  echo ${STATISTIC_TO_CHECK} | awk -F';' '{print $4}')
+    KB_REFERENCE=$(   echo ${STATISTIC_TO_CHECK} | awk -F';' '{print $5}')
+
+    # Only check DRIVES and COMPUTE for most of these
+    BACKEND_PROCESSES=$( ( weka cluster process -o id -F role=COMPUTE --no-header ; weka cluster process -o id -F role=DRIVES --no-header ) | paste -sd "," - )
```
Comment (vrragosta) [inline scripts.d/ta/780_statistical_outlier.sh:81]: Do we need to worry about exceeding a limit on the number of processes ids?

File: scripts.d/ta/780_statistical_outlier.sh [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+    eval $(weka stats --show-internal --stat ${STATISTIC_NAME} --per-process --interval ${INTERVAL} -s value --no-header -o  value -R --process-ids ${BACKEND_PROCESSES} | sed 's/[^0-9\.]//g' |  awk -f ${AWK_TEMP_FILE})
+    if (( $(echo "${HIGHEST_VALUE_SEEN} > (${STDDEV} * ${MAX_STDDEV})" |bc -l) )); then
+        echo "For the Weka container-level statistic ${STATISTIC_NAME} showed some outliers"
```
Comment (vrragosta) [inline scripts.d/ta/780_statistical_outlier.sh:85]: Remove **For**?

**Review by vrragosta — CHANGES_REQUESTED** [review]
Minor changes requested.

File: scripts.d/ta/780_statistical_outlier.sh (hunk 1 above)
Comment (jackchallen) [reply scripts.d/ta/780_statistical_outlier.sh:81]: I don't think so. There might be a shell command line length limit but again I don't /think/ it applies in shell script.

File: scripts.d/ta/780_statistical_outlier.sh (hunk 2 above)
Comment (jackchallen) [reply scripts.d/ta/780_statistical_outlier.sh:85]: fixed

File: scripts.d/ta/780_statistical_outlier.sh [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    INTERVAL=$(       echo ${STATISTIC_TO_CHECK} | awk -F';' '{print $2}')
+    MAX_STDDEV=$(     echo ${STATISTIC_TO_CHECK} | awk -F';' '{print $3}')
+    WTA_REFERENCE=$(  echo ${STATISTIC_TO_CHECK} | awk -F';' '{print $4}')
+    KB_REFERENCE=$(   echo ${STATISTIC_TO_CHECK} | awk -F';' '{print $5}')
+
+    # Only check DRIVES and COMPUTE for most of these
+    BACKEND_PROCESSES=$( ( weka cluster process -o id -F role=COMPUTE --no-header ; weka cluster process -o id -F role=DRIVES --no-header ) | paste -sd "," - )
+
+    eval $(weka stats --show-internal --stat ${STATISTIC_NAME} --per-process --interval ${INTERVAL} -s value --no-header -o  value -R --process-ids ${BACKEND_PROCESSES} | sed 's/[^0-9\.]//g' |  awk -f ${AWK_TEMP_FILE})
+    if (( $(echo "${HIGHEST_VALUE_SEEN} > (${STDDEV} * ${MAX_STDDEV})" |bc -l) )); then
```
[the remark this answers is not in the document]
Comment (jackchallen) [reply scripts.d/ta/780_statistical_outlier.sh]: Should be addressed in latest commit. Thanks.

**Review by vrragosta — APPROVED** [review]
A-OK

**Pull request merged**